exercises/object-oriented-programming/README.md

```diff
@@ -7,88 +7,238 @@
-- saving accounts get interest added (via the method you already wrote); 
-- current accounts get a letter sent if they are in overdraft.
+In your terminal, run the following command:
 
-Additionally, the `Bank` class requires methods for opening and closing accounts, and for paying a dividend into each account.
+```shell
+./mvnw clean test -Dtest=CurrentAccountTest,SavingsAccountTest
+```
+
+If you are on Windows, run this command instead:
+
+```bat
+mvnw.cmd clean test -Dtest=CurrentAccountTest,SavingsAccountTest
+```
+
+Your implementation is correct when all tests pass.
+
+### Stretch Goal
+
+If you have extra time in the session, or wish to experiment further, create a `Bank` class to manage accounts. Consider the following ideas:
+- storing a internal list of accounts. Remember that accounts in the list could be instances of the `Account` class, the `SavingsAccount` class, or the `CurrentAccount` class.
+- opening a new account, given a type and balance.
+- getting a list of account numbers held by the bank
+- getting an account, given an account number
+- closing an account, given an account number
+- displaying a report of all accounts held by the bank
+- paying a dividend to all accounts held by the bank
+- applying interest to all savings accounts
+- contacting all current account holders if their account is in overdraft.
+
+Update `App.java` to create a bank instance and then execute your bank's operations
+
+#### Notes
+There are no tests for the `Bank` class, so it's up to you how to implement it.
 
 ## :pushpin: Abstraction
 
-In this exercise you will develop a class hierarchy of shapes and write a program that computes the amount of paint
-needed to paint different objects. 
-The hierarchy will consist of a parent class `Shape` with three derived classes - `Sphere`, `Rectangle`, and `Cylinder`. 
-For the purposes of this exercise, the only attribute a shape will have is a `name` and the method of
-interest will be one that computes the area of the shape (surface area in the case of three-dimensional shapes).
+In this exercise you will develop a class hierarchy of shapes and write a program that computes the amount of paint needed to paint different objects. 
+The hierarchy will consist of a parent class `Shape` with three derived classes - `Sphere`, `Rectangle`, and `Cylinder`.
+
+For the purposes of this exercise, the only attribute a shape will have is a `name` and the method of interest will be one that computes the area of the shape (surface area in the case of three-dimensional shapes).
+
+### Part 1
+
+Inside the `shapes` package, create an abstract class `Shape` with the following members:
+- `abstract double getArea()` - returns the area of the shape
+- `String getName()` - returns the name of the shape
+
+In the same package, create a concrete class `Sphere` that extends `Shape`:
+- `Sphere(double radius)` - constructor
```

Comment:
No, because each derived class would define its name internally; that wouldn't be passed into its constructor. The student could choose to define a constructor for the base class and pass it into _that_, however.